src/gui/processing/qgsprocessingfeaturesourceoptionswidget.cpp

```diff
@@ -31,6 +31,7 @@ QgsProcessingFeatureSourceOptionsWidget::QgsProcessingFeatureSourceOptionsWidget
   mComboInvalidFeatureFiltering->addItem( tr( "Do not Filter (Better Performance)" ), QgsFeatureRequest::GeometryNoCheck );
   mComboInvalidFeatureFiltering->addItem( tr( "Skip (Ignore) Features with Invalid Geometries" ), QgsFeatureRequest::GeometrySkipInvalid );
   mComboInvalidFeatureFiltering->addItem( tr( "Stop Algorithm Execution When a Geometry is Invalid" ), QgsFeatureRequest::GeometryAbortOnInvalid );
+  mComboInvalidFeatureFiltering->addItem( tr( "Try to fix Invalid Geometry, otherwise skip it. May discard m-value and other attr." ), QgsFeatureRequest::GeometryFixInvalid );
```

Comment:
Here I think we should keep the string shorter, and then show a big obnoxious warning label with all the consequences of this option when it's set. But I'm happy to handle that post merge.

Reply:
Any suggestion regarding the string?